purchase_triple_discount/models/product_supplierinfo.py

```diff
@@ -40,4 +40,4 @@ def create(self, vals):
                 vals['product_tmpl_id'] in self.env.context['discount3_map']):
             vals['discount3'] = self.env.context['discount3_map'][
                 vals['product_tmpl_id']]
-        return super(ProductSupplierInfo, self).create(vals)
+        return super().create(vals)
```

Comment:
Maybe we should introduce here a hook in `purchase_discount` as well for not having to check the mapping again (and include all lines in the initial mapping)?

Reply:
Yes, thanks, good idea. What about this diff ?

https://github.com/OCA/purchase-workflow/pull/733/commits/0a670aa1547f4add0654f85478a8ebde8c7e8f05

regards.